spring-xd-spark-streaming-tests/src/test/java/org/springframework/xd/spark/streaming/KafkaTransportSparkStreamingTests.java

```diff
@@ -31,16 +27,11 @@
 public class KafkaTransportSparkStreamingTests extends AbstractSparkStreamingTests {
 
 	@ClassRule
-	public static KafkaTestSupport kafkaTestSupport;
+	public static KafkaTestSupport kafkaTestSupport = new KafkaTestSupport();
 
 	static {
-		kafkaTestSupport = new KafkaTestSupport();
-		Properties brokerConfigProperties = TestUtils.createBrokerConfig(0, TestUtils.choosePort());
-		kafkaTestSupport.setBrokerConfig(brokerConfigProperties);
-		System.setProperty(KafkaConnectionPropertyNames.KAFKA_BROKERS,
-				brokerConfigProperties.getProperty("host.name") + ":" + brokerConfigProperties.getProperty("port"));
-		System.setProperty(KafkaConnectionPropertyNames.KAFKA_ZK_ADDRESS,
-				brokerConfigProperties.getProperty("zookeeper.connect"));
+		System.setProperty(KafkaConnectionPropertyNames.KAFKA_BROKERS, kafkaTestSupport.getBrokerAddress());
+		System.setProperty(KafkaConnectionPropertyNames.KAFKA_ZK_ADDRESS, kafkaTestSupport.getZkConnectString());
 	}
```

Comment:
I think that the original intent of the code was to control how the `KafkaTestSupport` `@ClassRule` rule is instantiated, and to retrieve its configuration parameters, so that they can be set as system properties. Done this way, the instantiation has to take place in a static block, because `@ClassRule`s are executed around the `@BeforeClass`/`@AfterClass` cycle.

After moving the instantiation to the field declaration, there is little reason to keep a static block, that is correct.

Also, looking again at this, I notice that we need to move this into a `@BeforeClass` block, for the embedded tests to work properly.